connectors/src/connectors/github/temporal/activities.ts

```diff
@@ -711,3 +728,261 @@ export function getDiscussionDocumentId(
 ): string {
   return `github-discussion-${repoId}-${discussionNumber}`;
 }
+
+export function formatCodeContentForUpsert(
+  sourceUrl: string,
+  content: Buffer
+): CoreAPIDataSourceDocumentSection {
+  // For now we simply add the file name as prefix to all chunks.
+  return {
+    prefix: `SOURCE FILE: ${sourceUrl}\n\n`,
+    content: content.toString(),
+    sections: [],
+  };
+}
+
+export async function githubCodeSyncActivity(
+  dataSourceConfig: DataSourceConfig,
+  installationId: string,
+  repoLogin: string,
+  repoName: string,
+  repoId: string,
+  isBatchSync: boolean,
+  loggerArgs: Record<string, string | number>
+) {
+  const localLogger = logger.child(loggerArgs);
+
+  const connector = await Connector.findOne({
+    where: {
+      connectionId: installationId,
+      dataSourceName: dataSourceConfig.dataSourceName,
+      workspaceId: dataSourceConfig.workspaceId,
+    },
+  });
+  if (!connector) {
+    throw new Error(`Connector not found (installationId: ${installationId})`);
+  }
+
+  const connectorState = await GithubConnectorState.findOne({
+    where: {
+      connectorId: connector.id,
+    },
+  });
+  if (!connectorState) {
+    throw new Error(`Connector state not found for connector ${connector.id}`);
+  }
+
+  if (!connectorState.codeSyncEnabled) {
+    localLogger.info("Code sync disabled for connector");
+    return;
+  }
+
+  const { tempDir, files, directories } = await processRepository({
```

Comment:
Do we have any control over the max size of each tempDir ? I believe our pods get 1gb storage by default, but a single repo could already fill that up easily. If we have several concurrent syncs happening this could get filled up quite easily. 

Additionally, there's currently no concurrency control on github workflows, so it's also possible that several code changes in rapid succession for the same repo would lead to multiple concurrent activities competing for the `repo-xxx` tempdir. Not really an issue yet since the activity isn't called in a workflow yet, but something to keep in mind when working on the workflow code.

Reply:
I wonder if temporary GCP storage objects could potentially be a better fit than the actual file system of the pod ?

Reply:
We need to untar so we can't use GCP.

We limit ourselves to 2GB repo size, which means south of 1GB per tar ball. Plan is to increase pod storage to 32GB SSD

Reply:
SGTM